### ISSUE TITLE
Drop constructor() if it contains only super() with same arguments

### ISSUE THREAD

**Issue by rentalhost** [swc-project/swc#7336]
### Describe the feature

I have a `extends Error` class, but how TypeScript defines `cause` as `unknown`, I need just rewrite `constructor()` to set `cause` type as I expect. But when it is transformed into JavaScript, so it can be just dropped.

```ts
export class GenericError extends Error {
  public cause!: { value: number };

  public constructor(
    message: string,
    cause: { value: number }
  ) {
    super(message, cause);
  }
}
```

So, I can do that, with TS validation to `number` for `cause`:

```ts
throw new GenericError("message", { cause: 123 });
```

It is just transformed into (pretty-printed):

```js
export class GenericError extends Error{
  cause;
  constructor(r,e) {
    super(r,e)
  }
}
```

Once `constructor()` just calls `super()` with the same args, so I think it can be safetly dropped.

https://play.swc.rs/?version=1.3.55&code=H4sIAAAAAAAAA22NQQqDMBBF94J3%2BN0peAJdl54jpoMEYhJmMkUQ796k6aKLLuf9%2F9%2FQkSJnWG9E8KBA7OydOTLoyBSegnadfQckXb2zsEaFbjNOvIxXmhF0X4lxLX33W4tBMqvNkYeKgZ1EzFYGhbuwTY1%2BdH9sNR3bY0A0EQ9fwdQ241Kz0rveS6BdBMYAAAA%3D&config=H4sIAAAAAAAAA41VwW7bMAy99ysCn3fochiGfkBv%2BwZBsShHnSwaIuUmKPLvoxU7TRva2CWI%2BfRIkXykPp52u%2BaN2uZl9yF%2F5WOwmSDfvsVC58T2JJaGzwNQm8PAzY8FZZogbyNBNV2uSMM2d8ATC2j%2FvN%2FPjCYiEiyM2daHFPz5PmaL%2FZCB6M4mVnFZekhMX%2FkzlvF9AjiXe%2FsBMYJNG4ixZEJi6CBrjluM0Q4EZrRZ8TLd1OZAqIWYwMLgzJBxUPHkAgdMEvMRdWCdadGBAoUMLYcRNJrEEloiSU%2FJp8IODqXrap%2B%2FsWG0sVhWYsKptkRuq3g9YiA2viSthFdwpQZXcC7ud2bwJgOXnB55bxjSSk%2F%2BAkgFoiVKtgfNbz3hRU9rbL%2FJDMmLZPms4KJvLcsEnRTVhOCVyk6VgcxB62YGV1qYKttq15nhlfJRcGDAe9GK4preA7dHLeg06OgVQPprvaaqK2BuU7iCTwOxAb9KlqwLbD7RWz6uo3TuDxg3AvTAR3QbB6QVjOtwli1xGtbxkhyINMCpRwpV4HEJyAAwmlj35YM2ZDzEo%2BkiHj7XxHzgctvDvU1dnfe7fck4RBghrsn4P0ZkE5W7jZOql8l%2BmF74rYnSyuCFn89fng1J5Wn5rUk1Pbpyl1DV5PU5%2BdV8HlpejqVqTaA%2FC7HW6fIP6MYcrOMGAAA%3D

https://github.com/microsoft/TypeScript/issues/54021

**Comment by kdy1** [comment]
I don't expect this to have meaningful difference in real-world apps.

**Comment by rentalhost** [comment]
@kdy1 I found a lot of real-world examples based on same concept: to strong typing.

- https://grep.app/search?q=constructor%5Cs%2A%5C%28.%2B%5C%29%5Cs%2A%5C%7B%5Cs%2Asuper%5C%28.%2B%5C%29%5Cs%2A%3B%5Cs%2A%7D&regexp=true&filter[lang][0]=TypeScript

**Comment by rentalhost** [comment]
And if you come to reconsider this case, perhaps this can be applied under the same condition for **[parent methods](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDiCmA7eAnAlsAMgewObQG8AoaU6ABwFcAjEDaEPACjUQDNsAuaMRATwCUhEmVHBsiCNhDwAdI1wt22AQG4RpAL5FtRUJBgA5SgFtqqHPngAPAC5IAJjATJ0WPMNEB6L9ADCvNDm0Ei2aCjwIHzQDijY5OTwDjxs9ijQtii8EBwoJrIaFDR0wAzMrBzciKbmKELEoqIRtpQoiNAQlIko8uXKaoXauuKSttC4SKgYltAAvNDIAO5wk26WTAMTrtN4vYoAjABMAMxq0D7QhhjwRFtT7rh7TADkqHEoz2cXV8A3ehIQMbVMwWDzzJaXGqgxQDYG1SxPI6nVTnXw-G5w6FPV4od6fFEXABCYGSTHMwDAlAg8GgaDGi0gCzAuOwiySAiIQA)**:

```ts
class GenericLog {
    public log(info: any) {
        console.log(info);
    }
}

class NumberLog extends GenericLog {
    public log(info: number) {
        return super.log(info);
    }
}

const genericLog = new GenericLog();
genericLog.log(123); // Nice
genericLog.log('error'); // Nice

const numberLog = new NumberLog();
numberLog.log(123); // Nice
numberLog.log('error'); // Bad (because it was narrowed)
```

There are **[also cases](https://www.typescriptlang.org/play?#code/MYGwhgzhAEDiCmA7eAnAlsAgiVAXaA3gFDSnQAOArgEYgbRg4q4AUaiAZgPYBcDiATwCUfMIMIkyUxnjacuQgNySppFPFyUUiaLhSV4yqQF8ipoqEgwAcpQC21VNjzR4AD1xIAJjATJ0WEz4xFIA9KHQAMJi0I6uiLho6iAC0F4oXOTk8F4MHJ4ouihiENwodgB0KhQ0dMAMQXLcfIj2jigi-KkhqmoaWjoQlNkoFTLMTQpGZKbmllDQAMp67ADmzsyuHt6+SKgYG8HV4VGY1tYA8gAqsfBpGVk5ADS3wGCUEHdo+Oqa2jAANy4aFyiC4AHcqlIqLR6ONWOxmtAICtEKtOkCQRJeqQhiMxo1EVNqrMiBYuIgUdBVnsAgAZLiraAAXmgyHBcFpB0aSiINP8GAZqwJsgAjAAmADMSmgJ2sGHgfK5wCFIomAHJUBkUOqZXKFWTgBSqa0HKghSy2fAObYzShDixeab2qr4SwJdLFLKIvLgIrnebGWrWJqUNrdV6TgAhMC5FiON4fL74cGQNlgMMQnJCQ3G-Ao9Boi2s9lLVHrHnKAtrV2NUPhvU+g3VotBt0exvQGNxhPvT7Qb7QVMwRAZjLg7NEIA)** where there will be no `return`, but the rule can only be applied if the parent method also doesn't `return` a value (as it's a more complex case, maybe it's better to just ignore it, unless there is a reliable solution for that).

```ts
class GenericAlert {
    public alert(info: any): any {
        alert(info);
        return true;
    }
}

class NumberAlert extends GenericAlert {
    // Can be entirely dropped after transform.
    public alert(info: number): any {
        return super.alert(info);
    }
}

class StringAlert extends GenericAlert {
    // CANNOT be dropped, because it returns void now.
    public alert(info: string): void {
        super.alert(info);
    }
}

const genericLog = new GenericAlert();
genericLog.alert(123); // Nice
genericLog.alert('error'); // Nice

const numberLog = new NumberAlert();
numberLog.alert(123); // Nice
numberLog.alert('error'); // Bad (because it was narrowed)

const stringLog = new StringAlert();
stringLog.alert('error'); // Nice
stringLog.alert(123); // Bad (because it was narrowed)
```

**Comment by NullVoxPopuli** [comment]
this''d be really great to have!

**Comment by NullVoxPopuli** [comment]
Related:
- https://github.com/swc-project/swc/issues/7336
- https://github.com/swc-project/swc/issues/8177
- https://github.com/evanw/esbuild/issues/3463
- https://github.com/evanw/esbuild/issues/3464
- https://github.com/terser/terser/issues/1467
- https://github.com/terser/terser/issues/1468

**Comment by Austaras** [comment]
Actually the examples posted in this issue is not meaningless, consider:
```
class A {
    constructor(...args) {
        console.log(args)
    }
}

class B extends A {
    constructor(a) {
        super(a)
    }
}
```
if we call new B with 1, 2, 3, it would only log 1. But if we remove the seemingly useless constructor, it would log 1, 2, 3

**Comment by NullVoxPopuli** [comment]
Right, so constructor dropping needs to be aware of console dropping

**Comment by kdy1** [comment]
Sorry. It should not be implemented.
Minifier should never be aware of such thing
